gadgets/snapshot_socket/test/snapshot_socket_test.go

```diff
@@ -0,0 +1,116 @@
+// limitations under the License.
+package tests
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	gadgettesting "github.com/inspektor-gadget/inspektor-gadget/gadgets/testing"
+	igtesting "github.com/inspektor-gadget/inspektor-gadget/pkg/testing"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/containers"
+	igrunner "github.com/inspektor-gadget/inspektor-gadget/pkg/testing/ig"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/match"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/utils"
+	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
+)
+
+type snapshotSocketEntry struct {
+	eventtypes.CommonData
+
+	NetNsID     uint64 `json:"netns_id"`
+	InodeNumber uint64 `json:"ino"`
+
+	SrcEndpoint utils.L4Endpoint `json:"src"`
+	DstEndpoint utils.L4Endpoint `json:"dst"`
+	Status      string           `json:"status"`
+}
+
+func TestSnapshotSocket(t *testing.T) {
+	gadgettesting.RequireEnvironmentVariables(t)
+	utils.InitTest(t)
+
+	containerFactory, err := containers.NewContainerFactory(utils.Runtime)
+	require.NoError(t, err, "new container factory")
+	containerName := "test-snapshot-socket"
+	containerImage := "docker.io/library/busybox:latest"
+
+	var ns string
+	containerOpts := []containers.ContainerOption{containers.WithContainerImage(containerImage)}
+
+	if utils.CurrentTestComponent == utils.KubectlGadgetTestComponent {
+		ns = utils.GenerateTestNamespaceName(t, "test-snapshot-socket")
+		containerOpts = append(containerOpts, containers.WithContainerNamespace(ns))
+	}
+
+	testContainer := containerFactory.NewContainer(
+		containerName,
+		"nc -l 0.0.0.0 -p 9090",
+		containerOpts...,
+	)
+
+	testContainer.Start(t)
+	t.Cleanup(func() {
+		testContainer.Stop(t)
+	})
+
+	var runnerOpts []igrunner.Option
+	var testingOpts []igtesting.Option
+	commonDataOpts := []utils.CommonDataOption{utils.WithContainerImageName(containerImage), utils.WithContainerID(testContainer.ID())}
+
+	// TODO: timeout shouldn't be required
+	switch utils.CurrentTestComponent {
+	case utils.IgLocalTestComponent:
+		runnerOpts = append(runnerOpts, igrunner.WithFlags(fmt.Sprintf("-r=%s", utils.Runtime), "--timeout=1"))
+	case utils.KubectlGadgetTestComponent:
+		runnerOpts = append(runnerOpts, igrunner.WithFlags(fmt.Sprintf("-n=%s", ns), "--timeout=1"))
+		testingOpts = append(testingOpts, igtesting.WithCbBeforeCleanup(utils.PrintLogsFn(ns)))
+		commonDataOpts = append(commonDataOpts, utils.WithK8sNamespace(ns))
+	}
+
+	runnerOpts = append(runnerOpts, igrunner.WithValidateOutput(
+		func(t *testing.T, output string) {
+			expectedEntry := &snapshotSocketEntry{
+				CommonData: utils.BuildCommonData(containerName, commonDataOpts...),
+				SrcEndpoint: utils.L4Endpoint{
+					Addr:    "0.0.0.0",
+					Version: 4,
+					Port:    9090,
+				},
+				DstEndpoint: utils.L4Endpoint{
+					Addr:    "0.0.0.0",
+					Version: 4,
+					Port:    0,
+				},
+				Status:      "LISTEN",
```

Comment:
We don't have yet a [replacer](https://github.com/inspektor-gadget/inspektor-gadget/blob/b461a4d992832222e2ff9023e6848b3fb29e7885/pkg/operators/formatters/formatters.go#L205) for parsing the socket status as [we were doing in the built-in version](https://github.com/inspektor-gadget/inspektor-gadget/blob/b461a4d992832222e2ff9023e6848b3fb29e7885/pkg/gadgets/snapshot/socket/tracer/tracer.go#L50-L76).

That's another good first issue if you want to work on that 🙂.

Reply:
> We don't have yet a [replacer](https://github.com/inspektor-gadget/inspektor-gadget/blob/b461a4d992832222e2ff9023e6848b3fb29e7885/pkg/operators/formatters/formatters.go#L205) for parsing the socket status as [we were doing in the built-in version](https://github.com/inspektor-gadget/inspektor-gadget/blob/b461a4d992832222e2ff9023e6848b3fb29e7885/pkg/gadgets/snapshot/socket/tracer/tracer.go#L50-L76).
> 
> That's another good first issue if you want to work on that 🙂.

Definitely, I am looking forward to it 🚀.

Reply:
You need to set 10 (corresponding value for LISTEN) here and make the variable an integer in snapshotSocketEntry.

Reply:
> You need to set 10 (corresponding value for LISTEN) here and make the variable an integer in snapshotSocketEntry.

Done! But soon I will start working on replacer for parsing the socket status.

Reply:
I saw someone already create a PR for it: https://github.com/inspektor-gadget/inspektor-gadget/pull/3314, but you can still contribute by testing and reviewing the proposal provided in that PR.